src/libstd/alloc.rs

```diff
@@ -13,15 +13,77 @@
 #![unstable(issue = "32838", feature = "allocator_api")]
 
 #[doc(inline)] #[allow(deprecated)] pub use alloc_crate::alloc::Heap;
-#[doc(inline)] pub use alloc_crate::alloc::{Global, oom};
+#[doc(inline)] pub use alloc_crate::alloc::{Global, Layout, oom};
 #[doc(inline)] pub use alloc_system::System;
 #[doc(inline)] pub use core::alloc::*;
 
+use panicking::HOOK_LOCK;
```

Comment:
Is there any particular reason we're reusing this lock?

Reply:
I saw no particular reason *not* to reuse it.

Reply:
OOM hooks and panic hooks are totally unrelated concepts :P Let's avoid the overlap.